core/src/main/java/com/adobe/aem/commons/assetshare/content/AssetModel.java

```diff
@@ -19,52 +19,61 @@
 
 package com.adobe.aem.commons.assetshare.content;
 
+import com.day.cq.dam.api.Asset;
 import com.day.cq.dam.api.Rendition;
+import com.day.cq.dam.commons.util.DamUtil;
 import org.apache.sling.api.resource.Resource;
 import org.apache.sling.api.resource.ValueMap;
+import org.osgi.annotation.versioning.ProviderType;
 
 import java.util.List;
 
+@ProviderType
 public interface AssetModel {
     String OVERRIDE_RESOURCE = "overrideAsset";
 
     /**
      * @return the [dam:Asset] resource this Asset model represents.
      */
-    public Resource getResource();
+    Resource getResource();
 
     /**
      * @return the absolute JCR path to the Asset
      */
-    public String getPath();
+    String getPath();
 
     /**
      * @return the escaped Asset path for use in a URL
      */
-    default public String getUrl() { return getPath(); }
+    default String getUrl() { return getPath(); }
+
+    /**
+     * @return the underlying CQ Asset object
+     */
+    default Asset getAsset() { return DamUtil.resolveToAsset(getResource()); }
 
     /**
      * @return the Assets' Id ([dam:Asset]/jcr:contnet@jcr:uuid)
      */
-    public String getAssetId();
```

Comment:
@davidjgonzalez just curious about the motivation behind removing the public modifier?

Reply:
ahh - theyre just implicit on interfaces., adding them was redundant originally.